initializers/logger.js

```diff
@@ -25,16 +26,24 @@ const createGcLogger = logger => ({
 
 		return new Proxy(child, handler);
 	},
-	log: level => function() {
-		if(arguments.length) {
+	// this nested function is left as a un-named function so we have access to the arguments object
+	log: level => function () {
+		if (arguments.length) {
 			const [first, data = {}, meta = {}] = arguments;
-			let message, err;
+			let message;
+			let err;
 
-			if(typeof first === 'string') {
+			if (typeof first === 'string') {
 				message = first;
 			} else if (first && first.message) {
 				err = first;
-				message = data.supplementalMessage || err.message;
+				if (err.response && err.response.data) {
+					message = JSON.stringify(err.response.data)
+				} else if (err.response && err.response.body) {
+					message = `${err.response.statusMessage} ${err.response.body}`
+				} else {
+					message = data.supplementalMessage ? err.message + data.supplementalMessage : err.message;
```

Comment:
I think we want the suppementalMessage anytime it exists.  I'd determine the `message` with your if/else and then add on the supplementalMessage after that, when it exists.

Reply:
i'm kind of wary of overly-specialized logic inside of this function. bunyan has a concept of serializers that is supposed to encapsulate logging behavior for specific inputs, might be worth looking into making a custom serializer instead. also, i believe it may already provide one for certain types of http responses, might be able to just use that. 
https://github.com/trentm/node-bunyan#serializers

Reply:
Isn't that equally true of the `supplementalMessage`?  I'd move it to a serializer if we were conditionally including it.  If its the base behavior for the logger I'm not sure what we gain or lose by moving it to the a serializer.  Thoughts?

Reply:
Looking at the docs, I could make a custom serializer for response objects but from what I could tell I would also have to edit each call to the logger to include the `res`(or whatever we call it) property for the serializer to get used properly, like this `log.err({req: <request object>}, 'something about handling this request');` which I think kind of is what JD wanted to avoid by doing this. I don't think the built in HTTP serializer would work with the axios response objects.

Reply:
yeah that seems reasonable. have it all just be part of an error serializer

Reply:
I think a custom error serializer is a good long term solution, I think we'd still have to go change our calls to .error to look like `gcLogger.err({ err }, ...)` but I could be reading the docs wrong. I'd be open to doing it as a part of this though, or it could also be it's own task.

Reply:
looking at the docs more carefully, i think you're right, the logging call structure would have to change to use serializers. I'd rather optimize for better end user experience, so I like sticking with your implementation for now.

Reply:
actually, i take that back. if you look at line 62, where the actual call to the proxied bunyan logger happens, the data is put into an object that has an `err` prop if `err` is defined. so i don't think calls would have to change

Reply:
as for whether or not making a custom serializer should be part of this or another ticket, i'm not sure i feel qualified to make that decision ;)

Reply:
ah I see, yeah I think you're right, we could just add a custom serializer for err. It looks like we're already using the stdSerializers which includes one for err though. So I think our custom one would replace the standard one, not a big deal really, we'd just want to make sure our custom one still does what the standard one did or at least the parts we find useful. Looks like it's nothing fancy 
`var errSerializer = Logger.stdSerializers.err = function (err) {
    if (!err || !err.stack)
        return err;
    var obj = {
        message: err.message,
        name: err.name,
        stack: getFullErrorStack(err),
        code: err.code,
        signal: err.signal
    }
    return obj;
};`

Reply:
I think we'd still need custom logic for the supplemental message since it's not attached to the error object, or we could also do custom serializers for the data and meta arguments

Reply:
I think custom serializers for err, data, and meta would all be a good ideas, but probably best suited to their own ticket(s), unless that's something you'd want me to tackle now @bearfact

Reply:
sounds good to me